vendor/k8s.io/kubernetes/pkg/kubelet/kubelet.go

```diff
@@ -1242,7 +1242,7 @@ func (kl *Kubelet) tryRegisterWithApiServer(node *api.Node) bool {
 				// controller-managed attach detach annotation for this node.
 				requiresUpdate := kl.reconcileCMADAnnotationWithExistingNode(node, existingNode)
 				if requiresUpdate {
-					if _, err := kl.kubeClient.Core().Nodes().Update(existingNode); err != nil {
+					if _, err := kl.kubeClient.Core().Nodes().UpdateStatus(existingNode); err != nil {
```

Comment:
this line needs to go in a commit titled: `UPSTREAM: <carry>: Tolerate node ExternalID changes with no cloud provider`